lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -1474,8 +1474,17 @@ _LoadUdimTexture(const std::string& path, bool& isColorSpaceSRGB, MFloatArray& u
         return nullptr;
     }
 
+    // The UDIM scale and offset cannot be retrieved from a MTexture*, so we need to remember
+    // what was the result the last time we created one. This is required because a pointer
+    // in the global and local cache can go stale, while Maya still remembers about that
+    // texture.
```

Comment:
Found while playing around with UDIMs. It was very easy to break them. Just open the regular "stage_with_udims.usda" in Maya without this fix and duplicate the grid. This will cause the scale and offset to be lost and break the render.

Reply:
Should we clean kScaleOffsetCache somewhere? Should it really be static?

Reply:
Can only be cleaned if we know for sure the texture manager forgot about this texture. I do not see any API for this.  Should not get very big, and can not get stale since we only fetch the cached value if the MTextureManager found it.

Reply:
Would be nice if the information could be retrieved from the MTextureDescription returned by MTexture::getDescription().